shiny/express/_page.py

```diff
@@ -0,0 +1,334 @@
+from __future__ import annotations
+
+from typing import Callable, Literal, Optional, cast
+
+from htmltools import Tag, TagAttrValue, TagChild, TagList
+
+from .. import ui
+from ..types import MISSING, MISSING_TYPE
+from ..ui._navs import NavMenu, NavPanel
+from ..ui._sidebar import Sidebar
+from ..ui.css import CssUnit
+from ._recall_context import RecallContextManager
+from ._run import get_top_level_recall_context_manager
+
+__all__ = (
+    "page_auto",
+    "page_auto_cm",
+    "set_page_title",
+    "set_page_lang",
+    "set_page_fillable",
+    "set_page_wider",
```

Comment:
It still seems to me that we should have a singular `set_page_preferences()` function, rather than one function per page parameter. I can see two benefits of doing so:

* API surface doesn't need to expand for other parameters we want to bless (i.e., should we be including `set_page_gap`/`set_page_padding` for a `page_fillable()` context?)
* With this current API, it's not very clear that you're really setting _preferences_ (at least for `fillable`/`wider`) that may or may not effect the end result

Reply:
I agree that a function like `set_page_preferences()` could make sense for things like `fillable` and `wider`, but the title feels different to me. It's not a preference like the others, and it's such a common thing to do that this feels really verbose:

```py
layout.set_page_preferences(title = "My Title")
```

Reply:
Maybe:

```
layout.page_opts(
  title = "My title",
  lang = "en",
  filllable = True,
  full_width = True,
)
```

Reply:
This could also work in Shiny Core:

```py
ui.page(
  title = "My title",
  lang = "en",
  filllable = True,
  full_width = True,
)
```